.github/workflows/all-static-checks.yaml

```diff
@@ -61,3 +61,62 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/sweeps_workflow_verification.py
+    name: Static analysis
+  clang-tidy:
+    runs-on: ubuntu-latest
+    container: ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-amd64:latest
+    permissions:
+      pull-requests: write
+      # OPTIONAL: auto-closing conversations requires the `contents` permission
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+        submodules: "recursive"
+    - name: Set safe directory for Git
+      run: git config --global --add safe.directory /__w/tt-metal/tt-metal
```

Comment:
Is this a path TT defines?  Or GHA?  If it's GHA there should be a variable we can reference rather than a hardcoded path.